R/nm-join-bayes.R

```diff
@@ -480,7 +480,9 @@ sim_ipred <- function(mod, mod_mrgsolve, exts, data, join_col, y_col, pbar) {
         ipar_n <- dplyr::filter(ipar, .data$draw == ext_row$draw)
         mrgsolve::idata_set(mod_sim, ipar_n) %>%
           mrgsolve::smat(mrgsolve::as_bmat(ext_row, "SIGMA")) %>%
-          mrgsolve::mrgsim_df(obsonly = TRUE, carry_out = join_col) %>%
+          mrgsolve::mrgsim_df(
+            obsonly = TRUE, carry_out = join_col, etasrc = "idata.all"
```

Comment:
Now that `etasrc` is no longer the default "omega", is the `mrgsolve::zero_re(..., "omega")` call on line 471 still needed?

Reply:
Yeah ... that's not needed anymore.  I'm wondering about bringing in more of the inputs directly into the mrgsim call .... It could be more readable / consolidated like this?


```r
  rm(ipar_full)

  res <- future.apply::future_Map(
    function(ext, ipar) {
      `%>%` <- magrittr::`%>%`
      purrr::map(seq_len(nrow(ext)), function(n) {
        pbar("IPRED")
        ext_row <- ext[n, ]
        ipar_n <- dplyr::filter(ipar, .data$draw == ext_row$draw)
          mrgsolve::mrgsim_df(
            mod_mrgsolve,
            data = data, idata = ipar_n,
            sigma = mrgsolve::as_bmat(ext_row, "SIGMA"),
            obsonly = TRUE, carry_out = join_col,
            etasrc = "idata.all"
          ) %>%
          dplyr::select(all_of(join_col), DV_sim = all_of(y_col)) %>%
          dplyr::mutate(sample = n)
      })
    },
    exts, ipars,
    future.seed = TRUE,
    future.globals = FALSE
  )


```

Reply:
I like that suggestion.  I confirmed same result before and after with the script from my earlier [post](https://github.com/metrumresearchgroup/bbr.bayes/pull/118#issuecomment-1834189330) so I'd vote to go for it.

<details>
<summary>output</summary>

pre:

```
$ git rev-parse HEAD
1ac68da447ce05b080db24c1c5137cfd739c22b2

$ Rscript r.R
ℹ Loading bbr.bayes
Building 1100_mod ... done.
    EPRED  EPRED_lo  EPRED_hi     IPRED  IPRED_lo  IPRED_hi
 611.5181  296.7285 1153.5932  651.1202  400.0349  919.0930
```

post:

```diff
diff --git a/R/nm-join-bayes.R b/R/nm-join-bayes.R
index 233741c..9691427 100644
--- a/R/nm-join-bayes.R
+++ b/R/nm-join-bayes.R
@@ -468,9 +468,6 @@ sim_ipred <- function(mod, mod_mrgsolve, exts, data, join_col, y_col, pbar) {
   })
   rm(ipar_full)

-  mod_sim <- mrgsolve::zero_re(mod_mrgsolve, "omega") %>%
-    mrgsolve::data_set(data)
-
   res <- future.apply::future_Map(
     function(ext, ipar) {
       `%>%` <- magrittr::`%>%`
@@ -478,10 +475,12 @@ sim_ipred <- function(mod, mod_mrgsolve, exts, data, join_col, y_col, pbar) {
         pbar("IPRED")
         ext_row <- ext[n, ]
         ipar_n <- dplyr::filter(ipar, .data$draw == ext_row$draw)
-        mrgsolve::idata_set(mod_sim, ipar_n) %>%
-          mrgsolve::smat(mrgsolve::as_bmat(ext_row, "SIGMA")) %>%
           mrgsolve::mrgsim_df(
-            obsonly = TRUE, carry_out = join_col, etasrc = "idata.all"
+            mod_mrgsolve,
+            data = data, idata = ipar_n,
+            sigma = mrgsolve::as_bmat(ext_row, "SIGMA"),
+            obsonly = TRUE, carry_out = join_col,
+            etasrc = "idata.all"
           ) %>%
           dplyr::select(all_of(join_col), DV_sim = all_of(y_col)) %>%
           dplyr::mutate(sample = n)
```

```
$ Rscript r.R
ℹ Loading bbr.bayes
Building 1100_mod ... done.
    EPRED  EPRED_lo  EPRED_hi     IPRED  IPRED_lo  IPRED_hi
 611.5181  296.7285 1153.5932  651.1202  400.0349  919.0930
```

</details>

Reply:
I pushed the suggested changes with ffc915b.  Thanks.